src/v2/Artsy/Relay/middleware/cache/Cache.ts

```diff
@@ -0,0 +1,100 @@
+import { QueryResponseCache } from "relay-runtime"
+import createLogger from "v2/Utils/logger"
+import RelayQueryResponseCache from "relay-runtime/lib/network/RelayQueryResponseCache"
+
+const logger = createLogger("[v2/Artsy/middleware/cacheMiddleware]")
+
+export interface CacheConfig {
+  size: number
+  ttl: number
+}
+
+export class Cache {
+  isServer: boolean
+  cacheConfig: CacheConfig
+  relayCache: RelayQueryResponseCache
+
+  redisCache: {
+    get: (key: string) => Promise<string>
+    set: (key: string, value: string) => Promise<void>
+    del: (key: string) => Promise<void>
+    flushall: () => Promise<void>
+  }
+
+  constructor(cacheConfig: CacheConfig) {
+    this.isServer = typeof window === "undefined"
+    this.cacheConfig = cacheConfig
+    this.initRelayCache()
+    this.initRedisCache()
+  }
+
+  initRelayCache() {
+    this.relayCache = new QueryResponseCache(this.cacheConfig)
+  }
+
+  initRedisCache() {
+    if (!this.isServer) {
+      return
+    }
+
+    const { promisify } = require("util")
+    const redis = require("redis")
+    const client = redis.createClient()
```

Comment:
For illustration I wanted to keep this bare bones, but note we already have an (old) redis [cache util](https://github.com/artsy/force/blob/5c59333da35ad9b208c8da43dc8501cc158d819c/src/lib/cache.coffee#L17).